src/script/interpreter.cpp

```diff
@@ -1461,21 +1461,22 @@ static bool IsPushOnly(const Span<const unsigned char>& script)
     return true;
 }
 
-static bool VerifyWitnessProgram(const CScriptWitness& witness, int witversion, const std::vector<unsigned char>& program, unsigned int flags, const BaseSignatureChecker& checker, ScriptError* serror)
+static bool VerifyWitnessProgram(const CScriptWitness& witness, int witversion, const Span<const unsigned char>& program, unsigned int flags, const BaseSignatureChecker& checker, ScriptError* serror)
 {
     std::vector<std::vector<unsigned char> > stack;
-    CScript scriptPubKey;
+    CScript temp_script;
```

Comment:
As `temp_script` is only used once, it could be declared and initialized immediately (as `const`) below at the place where it is needed.

Reply:
That does not work (see other comment).